packages/plugin-apollo-client/src/engine/index.ts

```diff
@@ -0,0 +1,227 @@
+import {
+	ApolloClient,
+	ApolloError,
+	InMemoryCache,
+	useApolloClient,
+	useQuery as useApolloQuery,
+	type ApolloClientOptions,
+	type NormalizedCacheObject,
+	type OperationVariables,
+	type QueryHookOptions,
+	type QueryOptions,
+	type ServerError,
+	type ServerParseError,
+} from '@apollo/client';
+import { Logger } from '@snapwp/core';
+import { getGraphqlUrl } from '@snapwp/core/config';
+import { ApolloQueryProvider } from './query-provider';
+import type {
+	BaseQueryClientEngine,
+	fetchQueryArgs,
+	useQueryArgs,
+} from '@snapwp/types';
+
+export type clientType = ApolloClient< NormalizedCacheObject >;
+export type clientOptionsType = ApolloClientOptions< NormalizedCacheObject >;
+
+/**
+ * An adapter for Apollo Client that implements the BaseQueryClientEngine interface.
+ * This adapter provides methods for obtaining an Apollo Client instance, executing queries,
+ * and using queries as hooks.
+ */
+export class ApolloQueryClientEngine
```

Comment:
(Or ApolloClientAdapter)

```suggestion
export class ApolloClientEngine
```